subsquid/test/bondedFinanceTest.ts

```diff
@@ -0,0 +1,256 @@
+  name: "bondedFinance.offer",
+  method: "offer",
+  section: "bondedFinance",
+  versionInfo: "132",
+  era: { mortalEra: "0xb401" },
+  signer: "5yNZjX24n2eg7W6EVamaTXNQbWCwchhThEaSWB7V3GRjtHeL",
+  args: [
+    {
+      name: "offer",
+      type: '{"beneficiary":"AccountId32","asset":"u128","bondPrice":"u128","nbOfBonds":"u128","maturity":"ComposableTraitsBondedFinanceBondDuration","reward":"ComposableTraitsBondedFinanceBondOfferReward"}',
+      value: {
+        asset: 1,
+        reward: [Object],
+        maturity: [Object],
+        bondPrice: "0x0000000000000000016345785d8a0000",
+        nbOfBonds: 15,
+        beneficiary: ALICE_ADDRESS,
+      },
+    },
+    { name: "keepAlive", type: "bool", value: true },
+  ],
+  hash: "0x807a508abeda083d9cb6f7751329881f2d45981a82deb261f6aa19e37d3d3e63",
+  tip: BigInt(0),
+  indexInBlock: 2,
+};
+
+function createNewOfferEvent(offerId: string) {
+  let eventMock = mock(BondedFinanceNewOfferEvent);
+  let evt = {
+    offerId: BigInt(OFFER_ID_1),
+  };
+  when(eventMock.asV2300).thenReturn(evt);
+  when(eventMock.asLatest).thenReturn(evt);
+
+  let event = instance(eventMock);
+
+  return { event };
+}
+
+function createNewBondEvent(offerId: string, nbOfBonds: bigint) {
+  let eventMock = mock(BondedFinanceNewBondEvent);
+  let evt = {
+    offerId: BigInt(offerId),
+    who: WHO,
+    nbOfBonds,
+  };
+  when(eventMock.asV2300).thenReturn(evt);
+  when(eventMock.asLatest).thenReturn(evt);
+
+  let event = instance(eventMock);
+
+  return { event };
+}
+
+/**
+ * Check if bond offer has expected values
+ * @param bondArg
+ * @param id - Offer id
+ * @param purchased - Amount of purchased bonds
+ * @param beneficiary - Bond beneficiary
+ */
+function assertBondedFinanceBondOffer(
+  bondArg: BondedFinanceBondOffer,
+  id: string,
+  purchased: bigint,
+  beneficiary: string
+) {
+  expect(bondArg.id).to.equal(id);
+  expect(bondArg.purchased).to.equal(purchased);
+  expect(bondArg.beneficiary).to.equal(beneficiary);
+}
+
+/**
+ * @param ctx - Event context
+ * @param storeMock - Store mock
+ * @param offerId
+ */
+async function assertNewOfferEvent(
+  ctx: EventHandlerContext,
+  storeMock: Store,
+  offerId: string
+) {
+  const { event } = createNewOfferEvent(offerId);
+
+  await processNewOfferEvent(ctx, event);
+
+  // Check if the offer has been stored
+  const [arg] = capture(storeMock.save).last();
+  assertBondedFinanceBondOffer(
+    arg as unknown as BondedFinanceBondOffer,
+    offerId,
+    BigInt(0),
+    ALICE_ADDRESS
+  );
+}
+
+async function assertNewBondEvent(
+  ctx: EventHandlerContext,
+  storeMock: Store,
+  totalPurchased: Record<string, BondedFinanceBondOffer>,
+  totalPurchasedStored: Record<string, BondedFinanceBondOffer>,
+  offerId: string,
+  purchased: bigint
+) {
+  const { event } = createNewBondEvent(offerId, purchased);
+
+  // Update the total purchased bonds
+  totalPurchased[offerId].purchased += purchased;
+
+  await processNewBondEvent(ctx, event);
```

Comment:
I would move things above this line to the actual test, instead of this function as these are not assertions.

Reply:
done 👍🏼